### PR TITLE
circuits: zk-circuits: valid-match-settle: Add match unit tests

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
These tests are moved from the integration tests, they belong better as unit tests for testing circuit boundary cases. The integration tests will be used for connecting the match MPC with the circuit.

### Testing
- Unit tests pass

### CODE REVIEW

File: circuits/src/zk_circuits/valid_match_settle/mod.rs [hunk 1]
```diff
@@ -394,6 +395,174 @@ mod tests {
...
+        ));
+    }
+
+    /// Test the case in which the balance provided to the matching engine is
+    /// not for the correct asset
+    #[test]
+    fn test_valid_match__invalid_balance_mint() {
+        let (mut witness, statement) = dummy_witness_and_statement();
+
+        // Switch the mint of the balance to be the wrong asset in the pair
```
Comment (akirillo) [inline circuits/src/zk_circuits/valid_match_settle/mod.rs:463]:
```suggestion
        // Switch the mint of the balance to be an incorrect asset
```

File: circuits/src/zk_circuits/valid_match_settle/mod.rs [hunk 2]
```diff
@@ -394,6 +395,174 @@ mod tests {
         assert!(res.is_ok())
     }
 
+    // ---------------
+    // | Match Tests |
+    // ---------------
+
+    /// Randomly perform one of two operation
+    macro_rules! rand_branch {
```
Comment (akirillo) [inline circuits/src/zk_circuits/valid_match_settle/mod.rs:403]: this is nifty, nice

File: circuits/src/zk_circuits/valid_match_settle/mod.rs [hunk 3]
```diff
@@ -394,6 +395,174 @@ mod tests {
...
+        ));
+    }
+
+    /// Test the case in which the `max_minus_min` field is incorrectly computed
+    #[test]
+    fn test_valid_match__incorrect_max_minus_min() {
+        let mut rng = thread_rng();
+        let (mut witness, statement) = dummy_witness_and_statement();
+
+        // Add one to the max minus min amount
```
Comment (akirillo) [inline circuits/src/zk_circuits/valid_match_settle/mod.rs:506]:
```suggestion
        // Change the max minus min amount
```

File: circuits/integration/zk_circuits/valid_match_mpc.rs [hunk 4]
```diff
@@ -336,356 +336,8 @@ async fn test_valid_match__non_integral_price(test_args: IntegrationTestArgs) ->
...
-    if constraints_satisfied(witness, test_args.mpc_fabric.clone()).await? {
-        return Err(eyre!("Constraints satisfied on invalid witness"));
-    }
-
-    Ok(())
-}
-
-/// Test the case in which the `max_minus_min` field is switched to be
-/// `min - max` (i.e. negative)
-async fn test_valid_match__max_minus_min_negative(test_args: IntegrationTestArgs) -> Result<()> {
```
Comment (akirillo) [inline circuits/integration/zk_circuits/valid_match_mpc.rs:544]: Seems we didn't port over this test, do we not need it?

**Review by akirillo — APPROVED** [review]

File: circuits/integration/zk_circuits/valid_match_mpc.rs (hunk 4 above)
Comment (joeykraut) [reply circuits/integration/zk_circuits/valid_match_mpc.rs:544]: correct
